### DISCUSSION TITLE
Handling non-zero exit codes in exec

### DISCUSSION THREAD

**Discussion by pdvcs** [deepnoodle-ai/risor#241] [Q&A] [answered]
Hi everyone,

Thank you for building risor, it's a very cool project and I look forward to seeing what I can do with it.

I'm really interested in using Risor to add 'scriptability' to my Go programs, and thus far I was pleasantly surprised by how easy it is to interop between Risor and my own Go methods.

I did have a question which I was hoping you could help me with. Sorry if this is really basic.

I want to try and execute external programs using Risor -- but the thing is that sometimes these program invocations can fail for legitimate reasons. I'd like my script to be able to detect that -- and take corrective action where necessary.

For instance, consider

`exec("go", ["version"]).stdout // prints 'go version go1.22.3 ...'`

whereas

`exec("go", ["ionvers"]).stdout // 'exit status 2'`

Even try(...) doesn't help:

`try(exec("go", ["ionvers"]).stdout, 0) // 'exit status 2'`

This happens with exec.command(...) too:

`try(exec.command("go", "ionvers").combined_output()) // 'exit status 2'`

I looked at [exec/command.go](https://github.com/risor-io/risor/blob/main/modules/exec/command.go#L52-L54) and in general `modules/exec`, no workaround jumped out for me.

Essentially -- I couldn't figure out a way to handle exec 'failure' (i.e. non-zero exit codes). Is there a way or a workaround please?

Thanks!

**Comment by myzie** [comment, marked as answer]
Hi there @pdvcs, thanks for the feedback. I think someone else made a similar request recently:

https://github.com/risor-io/risor/issues/229

I agree this needs an adjustment and I should be able to get to it soon!

**Reply by pdvcs** [reply]
Thank you @myzie - worth noting that for my purposes this isn't a deal-breaker. I can simply call a method in my own Go code from risor if I need something the builtins / modules cannot provide. That makes it really well suited for embedding!
